common/variant/src/main/java/org/apache/spark/types/variant/Variant.java

```diff
@@ -316,6 +334,18 @@ static void toJsonImpl(byte[] value, byte[] metadata, int pos, StringBuilder sb,
       case BINARY:
         appendQuoted(sb, Base64.getEncoder().encodeToString(VariantUtil.getBinary(value, pos)));
         break;
+      case YEAR_MONTH_INTERVAL:
+        IntervalFields ymFields = VariantUtil.getYearMonthIntervalFields(value, pos);
+        int ymValue = (int) VariantUtil.getLong(value, pos);
```

Comment:
I'm not sure how this works. I thought the first byte at `pos` would include the start and end info. Then, I thought the remaining would be a 4 byte int? My confusion is:
- Shouldn't the `pos` read be `pos + 1`?
- Doesn't `VariantUtil.getLong(value, pos);` read variable length variant-encoded int? I thought we just need to read a 4-byte int, since the 4 bytes won't have any variant type info?

(similar question for `DAY_TIME_INTERVAL` below)